Zend/zend_language_scanner.l

```diff
@@ -1422,6 +1422,10 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	RETURN_TOKEN_WITH_IDENT(T_RETURN);
 }
 
+<ST_IN_SCRIPTING>"@@""\\"?{LABEL}("\\"{LABEL})* {
```

Comment:
The token should only be `@@`, the part after it should tokenize as a normal name.

Reply:
@nikic I do agree with that but the problem is that if `T_ATTRIBUTE` is changed to `"@@"` and the grammar adjusted to `attribute: T_ATTRIBUTE name` you will get a shift/reduce conflict in the parser due to ambiguity:
```php
function (@@Attr \stdClass $param) { }
```
Is this an attribute `Attr\stdClass` or an attribute `Attr` and a type-hint `\sdtClass`? The only solution I found was to make the name part of the token and deny use of whitespace within attribute names. Maybe there is a better way to accomplish that?

Reply:
Okay, this is a really big problem. This really needed to be part of the RFC. I'm afraid we will be forced to declare this proposal void on technical grounds.

Reply:
I don't think it makes sense to allow whitespace in identifiers. If https://wiki.php.net/rfc/namespaced_names_as_token is accepted then this won't be an issue.

Reply:
This was part of the original [Attributes v2 RFC](https://wiki.php.net/rfc/attributes_v2), which stated that the shorter `@:` syntax "does not permit whitespace in attribute names to allow detecting the ending of the declaration". However, I didn't think to mention that this applies to `@@` as well when I wrote the new RFC (the implementation is almost exactly the same).

But since attributes are new syntax anyway, I'm not sure it's a problem for them to have stricter semantics if necessary. Of course, the ideal solution would be to change the handling of namespaced names in general per https://wiki.php.net/rfc/namespaced_names_as_token, in which case there won't be any inconsistency.